python/lsst/afw/table/schema/schema.cc

```diff
@@ -0,0 +1,522 @@
+    cls.def("getBit", &Key<Flag>::getBit);
+}
+
+template <typename U>
+void declareKeySpecializations(PyKey<Array<U>> & cls) {
+    declareKeyAccessors(cls);
+    cls.def_property_readonly(
+        "subfields",
+        [](Key<Array<U>> const & self) -> py::object {
+            py::list result;
+            for (int i = 0; i < self.getSize(); ++i) {
+                result.append(py::cast(i));
+            }
+            return py::tuple(result);
+        }
+    );
+    cls.def_property_readonly(
+        "subkeys",
+        [](Key<Array<U>> const & self) -> py::object {
+            py::list result;
+            for (int i = 0; i < self.getSize(); ++i) {
+                result.append(py::cast(self[i]));
+            }
+            return py::tuple(result);
+        }
+    );
+}
+
+
+// Wrap all helper classes (FieldBase, KeyBase, Key, Field, SchemaItem) declarefor a Schema field type.
+template <typename T>
+void declareSchemaType(py::module & mod) {
+    std::string suffix = FieldBase<T>::getTypeString();
+    py::str pySuffix(suffix);
+
+    py::object astropyUnit = py::module::import("astropy.units").attr("Unit");
+
+    // FieldBase
+    PyFieldBase<T> clsFieldBase(mod, ("FieldBase" + suffix).c_str());
+    clsFieldBase.def_static("getTypeString", &FieldBase<T>::getTypeString);
+    declareFieldBaseSpecializations(clsFieldBase);
+
+    // KeyBase
+    PyKeyBase<T> clsKeyBase(mod, ("KeyBase" + suffix).c_str());
+    clsKeyBase.def_readonly_static("HAS_NAMED_SUBFIELDS", &KeyBase<T>::HAS_NAMED_SUBFIELDS);
+    declareKeyBaseSpecializations(clsKeyBase);
+
+    // Field
+    PyField<T> clsField(mod, ("Field" + suffix).c_str());
+    mod.attr("_Field")[pySuffix] = clsField;
+    clsField.def(
+        "__init__",
+        [astropyUnit]( // capture by value to refcount in Python instead of dangle in C++
+            Field<T> & self,
+            std::string const & name,
+            std::string const & doc,
+            py::str const & units,
+            py::object const & size,
+            py::str const & parse_strict
+        ) {
+            astropyUnit(units, "parse_strict"_a=parse_strict);
+            std::string u = py::cast<std::string>(units);
+            if (size == py::none()) {
+                new (&self) Field<T>(name, doc, u);
+            } else {
+                int s = py::cast<int>(size);
+                new (&self) Field<T>(name, doc, u, s);
+            }
+        },
+        "name"_a, "doc"_a="", "units"_a="", "size"_a=py::none(), "parse_strict"_a="raise"
+    );
+    clsField.def(
+        "_addTo",
+        [](Field<T> const & self, Schema & schema, bool doReplace) -> Key<T> {
+            return schema.addField(self, doReplace);
+        }
+    );
+    clsField.def("getName", &Field<T>::getName);
+    clsField.def("getDoc", &Field<T>::getDoc);
+    clsField.def("getUnits", &Field<T>::getUnits);
+    clsField.def("copyRenamed", &Field<T>::copyRenamed);
+    clsField.def(
+        "__repr__",
```

Comment:
Note that [DM-9140](https://jira.lsstcorp.org/browse/DM-9140) restores there being both `__str__` and `__repr__` for this class.